reactComponents/src/searchbar/tokenSearch/SearchIcon.tsx

```diff
@@ -0,0 +1,31 @@
+import React, {  memo } from 'react';
+import styled from 'styled-components';
+
+const StyledSVG = styled.svg`
+  '&:hover': {     
+      ${({ hoverColor}) =>
+        `stroke: ${hoverColor};`}
+     
+  }
+`
+export const SearchIcon = memo(
```

Comment:
do we get significant benefits for using memo?

Reply:
What's your opinion?

Reply:
its not a matter of opinion. https://blog.bitsrc.io/profiling-performance-of-react-apps-using-react-profiler-d02d77f3c96a

1. Do you see obvious slow down of the search component ?
2. If not, if you use React profile do you see bottlenecks in rerendering?
3. If you did see bottlenecks, once you use memo, did it improve the rerender speed? by how much?
If there is significant iprovements of loading speed then by by all means use memo. if not take it out

Reply:
unresolving this

Reply:
1. I don't see slow down of the search component with memo for icons.
2. I compared with Profiler extension, the component which is using the memo resource was blue color in flame chart, which means the component comparatively took less time to render
3. I don't see any bottleneck.

Reply:
if there is no significant decrease in render time when using use memo then dont use memo

Reply:
Ok, I will take memo out

Reply:
3. When to avoid React.memo()
If the component isn't heavy and usually renders with different props, most likely you don't need React.memo().

Use the following rule of thumb: don't use memoization if you can't quantify the performance gains.

Performance-related changes applied incorrectly can even harm performance. Use React.memo() wisely.

While possible, wrapping class-based components in React.memo() is undesirable. Extend PureComponent class or define a custom implementation of shouldComponentUpdate() method if you need memoization for class-based components.
i like eduardo's answer in his interview. He saw a problem. He found a component that does very expensive calculations when it rerenders (more than 5 seconds) Lets name that component "FOO" Then he finds any state/function/component that will make FOO rerender  and memoizes it.
https://stackblitz.com/edit/react-ts-xu667e?file=App.tsx
in this example, the App Component will cause the ComplicatedComponent to rerender when we update the App state. In this scenario, we would want the ComplicatedComponent not to rerender if the ComplicatedComponent does very heavy calculations inside it.
https://github.com/Rome-Blockchain-Labs/terminal-library/pull/12#discussion_r850365703

Reply:
@iandjx Great point, thanks.